extensions/kubernetes-client/deployment/pom.xml

```diff
@@ -25,6 +25,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-jackson-deployment</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.fabric8</groupId>
+            <artifactId>kubernetes-httpclient-okhttp</artifactId>
+            <scope>provided</scope>
```

Comment:
Why is the dependency `provided`?

Reply:
And why is it in the `deployment` module?

Maybe you're trying to make it optional? But I think we should make it runtime and non-optional and then switch to the vertx one as soon as it is ready, given that's our ultimate goal.

Reply:
The reasoning is:
- `kuberentes-httpclient-okhttp` is a `runtime` dependency of the `kubernetes-client` module. So for the `deployment` module, at runtime, the jar package with the SPI will always be available. Hence the `provided` scope.
- It's not an optional package because it's required at runtime.
- I added this as provided so I could simply import the factory in the statement:
  ```
  serviceProviderProducer.produce(new ServiceProviderBuildItem(
                "io.fabric8.kubernetes.client.http.HttpClient$Factory", OkHttpClientFactory.class.getName()));
  ```
  The alternative would be to hardcode the name of the class, but I think this would be less maintainable.

Please, let me know how do you want me to proceed.

Reply:
The semantics of `provided` in a `deployment` module are weird, so let's avoid them and do:

> The alternative would be to hardcode the name of the class

The native tests will pick up any breakage if for whatever version the class name changes in the future

Reply:
Sounds good.
I'll rebase and update.

Reply:
I might miss something but if `kubernetes-httpclient-okhttp` is a transitive dependency of the runtime module, then it should be around for the deployment module, which depends on the runtime one.

Reply:
It's a **runtime** the dependency of the runtime module :)

Anyway, it's changed now.